pkg/persistentcache/persistentcache.go

```diff
@@ -0,0 +1,61 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2019 Datadog, Inc.
+
+package persistentcache
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+)
+
+// Return a file where to store the data. We split the key by ":", using the
+// first prefix as directory, if present. This useful for integrations, which
+// use the check_id formed with $check_name:$hash
+func getFileForKey(key string) (string, error) {
+	parent := config.Datadog.GetString("run_path")
+	paths := strings.SplitN(key, ":", 2)
+	if len(paths) == 1 {
+		// If there is no colon, just return the key
+		return filepath.Join(parent, paths[0]), nil
+	}
+	// Otherwise, create the directory with a prefix
+	err := os.MkdirAll(filepath.Join(parent, paths[0]), 0700)
+	if err != nil {
+		return "", err
+	}
+	// Make the file Windows compliant
+	cleanedPath := strings.Replace(paths[1], ":", "_", -1)
```

Comment:
The full list is :
```
< (less than)
> (greater than)
: (colon)
" (double quote)
/ (forward slash)
\ (backslash)
| (vertical bar or pipe)
? (question mark)
* (asterisk)
```
Again, if there is any chance part of the key will be fetched dynamically, let's make sure it's sanitized

Reply:
https://github.com/DataDog/datadog-agent/blob/fdad756904022aaeb810eb33114d733b20d72d8a/tasks/test.py#L261